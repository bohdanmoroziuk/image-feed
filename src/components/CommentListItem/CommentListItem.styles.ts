import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    marginLeft: 20,
    paddingVertical: 20,
    paddingRight: 20,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: 'rgba(0, 0, 0, 0.05)',
  },
});
