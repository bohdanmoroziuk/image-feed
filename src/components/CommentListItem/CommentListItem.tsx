import { FunctionComponent } from 'react';
import { View, Text } from 'react-native';

import { Comment } from 'src/types';

import styles from './CommentListItem.styles';

export interface CommentListItemProps {
  item: Comment;
}

export const CommentListItem: FunctionComponent<CommentListItemProps> = ({ item }) => (
  <View style={styles.container}>
    <Text>
      {item.text}
    </Text>
  </View>
);
