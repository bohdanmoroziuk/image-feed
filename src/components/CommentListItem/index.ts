import { CommentListItem } from './CommentListItem';

export type { CommentListItemProps } from './CommentListItem';
export default CommentListItem;
