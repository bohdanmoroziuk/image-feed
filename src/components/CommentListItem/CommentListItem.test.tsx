import { render } from '@testing-library/react-native';

import { CommentListItem } from './CommentListItem';

describe('CommentListItem', () => {
  it('renders successfully', () => {
    const { toJSON, getByText } = render(
      <CommentListItem
        item={{
          id: '1',
          text: 'Awesome'
        }}
      />
    );

    expect(toJSON()).toMatchSnapshot();
    expect(getByText('Awesome')).toBeDefined();
  });
});
