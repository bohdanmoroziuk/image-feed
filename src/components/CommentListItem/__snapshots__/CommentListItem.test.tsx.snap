// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentListItem renders successfully 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "rgba(0, 0, 0, 0.05)",
      "borderBottomWidth": 0.5,
      "marginLeft": 20,
      "paddingRight": 20,
      "paddingVertical": 20,
    }
  }
>
  <Text>
    Awesome
  </Text>
</View>
`;
